server/api/auth/authController.js

```diff
@@ -0,0 +1,35 @@
+var createUser = require('../user/userController').create;
+var passport = require('passport');
+var jwt = require('jsonwebtoken');
+var config = require('../../config/development');
+
+exports.signup = function(req, res, next) {
```

Comment:
You can send signup as a passport strategy as well. Create the user, then authenticate if needed.